packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -133,6 +135,7 @@ export class RunningSummarizer implements IDisposable {
 	private heuristicRunner?: ISummarizeHeuristicRunner;
 	private readonly generator: SummaryGenerator;
 	private readonly mc: MonitoringContext;
+	private readonly shouldRestart: boolean;
```

Comment:
shouldAbortOnSummaryFailure? To make it explicit that in case of summary failures, this client will abort. The consequence is that a new summarizer will be chosen. It's not really a restart.